src/api/cpp/cvc5_checks.h

```diff
@@ -867,20 +892,8 @@ class CVC5ApiUnsupportedExceptionStream
     size_t size = bound_vars.size();                                          \
     CVC5_API_ARG_SIZE_CHECK_EXPECTED(size == domain_sorts.size(), bound_vars) \
         << "'" << domain_sorts.size() << "'";                                 \
-    size_t i = 0;                                                             \
-    for (const auto& bv : bound_vars)                                         \
+    for (size_t i = 0; i < size; ++i)                                         \
     {                                                                         \
-      CVC5_API_ARG_AT_INDEX_CHECK_NOT_NULL(                                   \
-          "bound variable", bv, bound_vars, i);                               \
-      CVC5_API_ARG_AT_INDEX_CHECK_EXPECTED(                                   \
-          d_tm.d_nm == bv.d_tm->d_nm, "bound variable", bound_vars, i)        \
-          << "a term associated with the term manager of this solver object"; \
-      CVC5_API_ARG_AT_INDEX_CHECK_EXPECTED(                                   \
-          bv.d_node->getKind() == cvc5::internal::Kind::BOUND_VARIABLE,       \
-          "bound variable",                                                   \
-          bound_vars,                                                         \
-          i)                                                                  \
-          << "a bound variable";                                              \
```

Comment:
Why are these checks no longer required? Did you mean to replace them with:
```
CVC5_API_SOLVER_CHECK_BOUND_VAR_AT_INDEX(bound_vars[i], bound_vars, i);
```
but you didn't?

Reply:
Previously, we performed all the checks for bound vars of define funs with this function. However, this happened at a stage that is too late for the future, when the NM is not a singleton anymore. That's why now these checks in the loop when we collect the domain sorts, and don't have to be repeated when this function is called. Do you want me to rename this function to reflect this? Maybe something like `CVC5_API_SOLVER_CHECK_BOUND_VARS_DEF_FUN_SORTS`?

Reply:
More importantly than renaming the function, which is fine, you should update its documentation, which still says:
```
* Check if each term in the given container is not null, associated with the
 * term manager of this solver, a bound variable, matches theh corresponding
 * sort in 'domain_sorts', and is a first-class term.
```
You should consider removing any parts that are no longer relevant, and adding a comment explaining why it's no longer necessary to check for bound variables.